test/gtest/pad_constant.hpp

```diff
@@ -0,0 +1,215 @@
+        // Generate random padding
+        for(size_t& i : padding)
+        {
+            i = prng::gen_descreet_unsigned<size_t>(1, 5);
+        }
+
+        std::vector<size_t> out_dims;
+        for(size_t i = 0; i < 5; i++)
+        {
+            out_dims.push_back(in_dims[i] + 2 * padding[2 * i]);
+        }
+
+        output = tensor<T>{out_dims};
+        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+        output_dev = handle.Write(output.data);
+
+        backward_output = tensor<T>{in_dims, strides};
+        std::fill(
+            backward_output.begin(), backward_output.end(), std::numeric_limits<T>::quiet_NaN());
+        backward_output_dev = handle.Write(backward_output.data);
+
+        ref_output = tensor<T>{out_dims};
+        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+    };
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        auto out_dims = output.desc.GetLengths();
+
+        float padding_value = 3.5f;
+
+        cpu_pad_constant_fwd<T>(input.data.data(),
+                                ref_output.data.data(),
+                                &input.desc,
+                                &output.desc,
+                                padding,
+                                padding_value);
+        miopenStatus_t status;
+
+        status = miopen::PadConstantForward(handle,
+                                            input.desc,
+                                            output.desc,
+                                            input_dev.get(),
+                                            output_dev.get(),
+                                            padding,
+                                            padding_value);
+        EXPECT_EQ(status, miopenStatusSuccess);
+        output.data = handle.Read<T>(output_dev, output.data.size());
+
+        status = miopen::PadConstantBackward(handle,
+                                             backward_output.desc,
+                                             output.desc,
+                                             backward_output_dev.get(),
+                                             output_dev.get(),
+                                             padding);
+        EXPECT_EQ(status, miopenStatusSuccess);
+
+        backward_output.data = handle.Read<T>(backward_output_dev, backward_output.data.size());
+
+        hipDeviceSynchronize();
+    }
+
+    void Verify()
+    {
+        // printf("Forward pass verification\n");
+        auto error = miopen::rms_range(ref_output, output);
+        EXPECT_TRUE(miopen::range_distance(ref_output) == miopen::range_distance(output));
+        EXPECT_TRUE(error == 0) << "Outputs do not match each other. Error:" << error;
+
+        // printf("Backward pass verification\n");
+        error = miopen::rms_range(backward_output, input);
+        EXPECT_TRUE(miopen::range_distance(backward_output) == miopen::range_distance(input));
```

Comment:
Well, my idea was that the code in `RunTest()` take the output of the forward pass as `input_grad` of the backward pass. To my understanding, in this case, it should cause the backward pass to create the same value as `input`. I can change it if necessary

Reply:
I think there can be various ways to verify the results. If you verify it logically like you said, please leave a comment with the logic. This is because the current code is difficult to understand intuitively.

Reply:
Okay, I've added the necessary notes